### ISSUE TITLE
False Positive Errors(?) When Passing `**kwargs: Unpack[...]` Typed Functions to `ParamSpec` Typed Higher-Order Functions

### ISSUE THREAD

**Issue by junkmd** [microsoft/pyright#6057]
**Describe the bug**
When passing the function `foo`, which uses `TypedDict` `Unpack`ed as a type annotation for `**kwargs`, to the higher-order function `call` that is typed with `ParamSpec`, it detects errors that might be false positives.

In the original function, reordering the keyword-only arguments or using keywords annotated with `**kwargs: Unpack[...]` does not result in an error. However, when using the keywords specified with `**kwargs: Unpack[...]` in the higher-order function, it raises an error.

**Code or Screenshots (actual behavior)**

```python
from collections.abc import Callable
from typing import Any, TypedDict, Unpack, TypeVar, ParamSpec


class _Kws(TypedDict, total=False):
    e: str
    f: str


_P = ParamSpec("_P")
_R = TypeVar("_R")


def foo(a: int, b: int, /, *, c: str = ..., d: str = ..., **kwargs: Unpack[_Kws]) -> float:
    ...


def call(func: Callable[_P, _R], *args: _P.args, **kwargs: _P.kwargs) -> _R:
    ...


foo(1, 2, e="bacon", c="spam")  # Valid
foo(1, 2, c="spam", d="ham", e="bacon")  # Valid
foo(1, 2, e="bacon")  # Valid


# `reportGeneralTypeIssues` are raised.
# Those said that ...
# `Argument of type "Literal['bacon']" cannot be assigned to parameter of type "*_Kws"`
# `"Literal['bacon']" is incompatible with "*_Kws"`
call(foo, 1, 2, e="bacon", c="spam")
#                 ^^^^^^^ Invalid
call(foo, 1, 2, c="spam", d="ham", e="bacon")
#                                    ^^^^^^^ Invalid
call(foo, 1, 2, e="bacon")
#                 ^^^^^^^ Invalid

```

**Expected behavior**
No error is raised.

**Additional context**
As VS Code extension integrated into Pylance `v2023.9.30`
On Python 3.11.4

**Comment by erictraut** [comment]
Yes, the code path that validates arguments for specialized param specs doesn't currently support unpacking of TypedDicts with a `**kwargs`. I didn't think it was likely that anyone would ever combine these two features, so I didn't bother adding it at the time. I'll look at adding this support.

**Comment by erictraut** [comment]
I took this opportunity to remove a bunch of redundant code in ParamSpec processing. Pyright now uses common logic, which means that all of the features that are supported for normal function calls are also supported when a ParamSpec is used. That includes PEP 692 unpacked TypedDict for `**kwargs` parameters.

This enhancement will be included in the next release.

**Comment by junkmd** [comment]
Thank you for your quick response!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.330, which I just published. It will also be included in an upcoming release of pylance.
